### ISSUE TITLE
repeated bytes when handling implicit tags

### ISSUE THREAD

**Issue by processor2018** [wbond/asn1crypto#97]
```python
import asn1crypto.core
import unittest

class SpcPeImageFlags(asn1crypto.core.BitString):
    _map = {
    0 : "includeResources",
    1 : "includeDebugInfo",    
    2 : "includeImportAddressTable",
    }

class SpcSerializedObject(asn1crypto.core.Sequence):
    _fields = [
        ("classId", asn1crypto.core.OctetString),
        ("serializedData", asn1crypto.core.OctetString),
    ]

class SpcString(asn1crypto.core.Choice):
    _alternatives = [
        ("unicode", asn1crypto.core.BMPString, {"implicit": 0}),
        ("ascii", asn1crypto.core.IA5String, {"implicit": 1}),
    ]

class SpcLink(asn1crypto.core.Choice):
    _alternatives = [
        ("url", asn1crypto.core.IA5String, {"implicit": 0}),
        ("moniker", SpcSerializedObject, {"implicit": 1}),
        ("file", SpcString, {"explicit": 2})
    ]

class SpcPeImageData(asn1crypto.core.Sequence):
    _fields = [
        ("flags",SpcPeImageFlags,{"default":"includeResources"}),
        ("file",SpcLink,{"explicit": 0})
    ]


class CoreTests(unittest.TestCase):
    def test_what_problem(self):
        bytes_piece = b"\x30\x09\x03\x01\x00\xA0\x04\xA2\x02\x80\x00"
        asn_piece = SpcPeImageData.load(bytes_piece, strict=True)
        self.assertEqual(bytes_piece[2:5], 
            asn_piece['flags'].dump())
        self.assertEqual(bytes_piece[5:11], 
            asn_piece['file'].dump())
 ```

**Comment by processor2018** [comment]
[test.txt](https://github.com/wbond/asn1crypto/files/2075421/test.txt)
Formatted test.py has been uploaded.

**Comment by wbond** [comment]
I did a little looking, and this seems to be something related to re-using data when dumping.

For instance, force a fresh dump fixes the test:

```python
class CoreTests(unittest.TestCase):
    def test_what_problem(self):
        bytes_piece = b"\x30\x09\x03\x01\x00\xA0\x04\xA2\x02\x80\x00"
        asn_piece = SpcPeImageData.load(bytes_piece, strict=True)
        self.assertEqual(bytes_piece[2:5], 
            asn_piece['flags'].dump())
        self.assertEqual(bytes_piece[5:11], 
            asn_piece['file'].dump(True))
```

**Comment by wbond** [comment]
This is fixed by bd66c499a2bc06c3b2551e279ba93f1256fc76fa. It was caused by the implementation of handling explicit headers and nested `core.Choice` objects.
